llvm/lib/Transforms/Vectorize/VPlanRecipes.cpp

```diff
@@ -2849,6 +2857,19 @@ InstructionCost VPExpressionRecipe::computeCost(ElementCount VF,
     Opcode = Instruction::Sub;
     LLVM_FALLTHROUGH;
   case ExpressionTypes::ExtMulAccReduction: {
+    if (isa<VPPartialReductionRecipe>(ExpressionRecipes.back())) {
+      auto *Ext0R = cast<VPWidenCastRecipe>(ExpressionRecipes[0]);
+      auto *Ext1R = cast<VPWidenCastRecipe>(ExpressionRecipes[1]);
```

Comment:
Does this work as expected for all test on current main? I think at least in some cases one of the operands may be a constant live-in.

Reply:
Yeah the matching function in `VPlanTransforms` explicitly checks for two extends, so the constant variant doesn't get bundled currently. I'm happy to get that working separately if necessary.